spec.html

```diff
@@ -8505,6 +8527,21 @@ <h1>OrdinaryFunctionCreate ( _functionPrototype_, _ParameterList_, _Body_, _this
         1. Set _F_.[[FormalParameters]] to _ParameterList_.
         1. Set _F_.[[ECMAScriptCode]] to _Body_.
         1. If the source text matching _Body_ is strict mode code, let _Strict_ be *true*; else let _Strict_ be *false*.
+        1. Set _F_.[[PresentInStackTraces]] to PresentInStackTraces of _Body_.
```

Comment:
Is the idea here to mirror the inheritance rules implicit in "is strict mode" but in a way that's simpler since these directives don't affect where future reserved words are allowed?

Reply:
That's correct. Since we're not introducing any parse errors or early errors, we can implement it using the already-existing spec machinery for runtime behaviour. From [the README](https://github.com/tc39/proposal-function-implementation-hiding#the-solution):

> Similar to the "use strict" directive, these new directives apply "inclusively downward", so that everything within the scope, plus the function itself when in function scope, gets hidden.